docs/proposals/configuration_changes.md

```diff
@@ -0,0 +1,71 @@
+## Configuration Changes
+
+### Goal
+Create a package for creating and managing configuration values as a value store. This will be used to remove the necessity of each package to be loaded just for configuration values. The other benefit is that this package could be a more robust solution to the configuration, Allowing us to do a better job of watching for updates to the configmap and managing subscriptions to those changes. This value store should also be usable by other parts of the application and therefore will require less usage of hard coded configuration structures and will allow the addition or removal of configuration values to not create such an issue. 
+
+### Work Items
+* Create a specific package for managing configuration 
+* Create a package mutex map to be used to store configuration values.
+* Take structured yaml and create a map of map of map.... saving the key as the name of the config value
+```go
+m["broker"] = map["dev_broker": true...]
+m["registry"] = [map["name": "dh"...], map["name": "play"...]]
+```
+* Create a syntax, and document, for retrieving a single configuration value from the list. I.E `config.MustGetBoolConfig("broker.dev_broker")`
+* Create a public API that will give the caller multiple ways to retrieve data.
+    - `MustGet` -> returns a single value and will panic if not able to retrieve the data
+    - `MustGet/Get<Type>` -> Allows for typed returns so the caller does not have to worry about type casting.
+    - Ability to specify a `map[string]interface{}` as a return value so one could say `config.MustGetMapConfig("broker")` and would retrieve the map of values for the broker configuration section. 
```

Comment:
I like that we can retrieve sections of the config info. Do you have an example of how we can use this? If you don't it's not a big deal because this can come in handy down the road.  My thought is this could be useful when we handle upgrading the broker.